sys/kern/syscalls.master

```diff
@@ -230,26 +258,26 @@
 		int ptrace(
 		    int req,
 		    pid_t pid,
-		    _Inout_opt_ caddr_t addr,
+		    _Inout_opt_ _Contains_ptr_ caddr_t addr,
```

Comment:
Why does ptrace addr requires Contrains_ptr annotation?

Reply:
`long` might be more accurate here, but if FreeBSD ptrace add definitely points to things containing pointers.

Reply:
Upon further reflection `_Contains_long_ptr_` is the correct annotation for addr since it could point to a struct ptrace_io_desc which contains both.